lib/activerecord-multi-tenant/migrations.rb

```diff
@@ -43,12 +43,45 @@ module ActiveRecord
   module ConnectionAdapters # :nodoc:
     module SchemaStatements
       alias :orig_create_table :create_table
+      alias :orig_change_table :change_table
+
+      def change_primary_key(table_name, options)
+        if !options[:partition_key] || options[:partition_key].to_s == 'id'
+          return
+        end
+
+        pkey_columns = ['id', options[:partition_key]]
+
+        # we are here comparing the columns in the primary key on the database and the one in the migration file
+        columns_result = execute("select kcu.column_name as key_column " \
+                                 "from information_schema.table_constraints tco "\
+                                 "join information_schema.key_column_usage kcu " \
+                                 "ON kcu.constraint_name = tco.constraint_name " \
+                                 "AND kcu.constraint_schema = tco.constraint_schema "\
+                                 "WHERE tco.constraint_type = 'PRIMARY KEY' " \
+                                 "AND tco.constraint_name = '#{table_name}_pkey'")
```

Comment:
Can we use a bind parameter here, instead of string substitution?

Whilst its unlikely that someone will do SQL injection here, it seems better to avoid problems with table_name called `'; DROP TABLE xyz; SELECT '`